test/payment/RefundEscrow.test.js

```diff
@@ -0,0 +1,114 @@
+      await this.contract.refund(investor1).should.be.rejectedWith(EVMRevert);
+    });
+
+    it('does not allow beneficiary withdrawal', async function () {
+      await this.contract.invest(investor1, { from: investor1, value: amount });
+      await this.contract.withdraw().should.be.rejectedWith(EVMRevert);
+    });
+  });
+
+  it('only owner can enter closed state', async function () {
+    await this.contract.close({ from: beneficiary }).should.be.rejectedWith(EVMRevert);
+
+    const receipt = await this.contract.close({ from: owner });
+
+    receipt.logs.length.should.equal(1);
+    receipt.logs[0].event.should.equal('Closed');
+  });
+
+  context('closed state', function () {
+    beforeEach(async function () {
+      await Promise.all(investors.map(investor => this.contract.invest(investor, { from: investor, value: amount })));
+
+      await this.contract.close({ from: owner });
+    });
+
+    it('rejects investments', async function () {
+      await this.contract.invest(investor1, { from: investor1, value: amount }).should.be.rejectedWith(EVMRevert);
+    });
+
+    it('does not refund investors', async function () {
+      await this.contract.refund(investor1).should.be.rejectedWith(EVMRevert);
+    });
+
+    it('allows beneficiary withdrawal', async function () {
+      const beneficiaryInitialBalance = await web3.eth.getBalance(beneficiary);
+      await this.contract.withdraw();
+      const beneficiaryFinalBalance = await web3.eth.getBalance(beneficiary);
+
+      beneficiaryFinalBalance.sub(beneficiaryInitialBalance).should.be.bignumber.equal(amount * investors.length);
+    });
+  });
+
+  it('only owner can enter refund state', async function () {
+    await this.contract.enableRefunds({ from: beneficiary }).should.be.rejectedWith(EVMRevert);
+
+    const receipt = await this.contract.enableRefunds({ from: owner });
+
+    receipt.logs.length.should.equal(1);
+    receipt.logs[0].event.should.equal('RefundsEnabled');
+  });
+
+  context('refund state', function () {
+    beforeEach(async function () {
+      await Promise.all(investors.map(investor => this.contract.invest(investor, { from: investor, value: amount })));
+
+      await this.contract.enableRefunds({ from: owner });
+    });
+
+    it('rejects investments', async function () {
+      await this.contract.invest(investor1, { from: investor1, value: amount }).should.be.rejectedWith(EVMRevert);
+    });
+
+    it('refunds investors', async function () {
+      for (let investor of [investor1, investor2]) {
+        const investorInitialBalance = await web3.eth.getBalance(investor);
+        const receipt = await this.contract.refund(investor);
+        const investorFinalBalance = await web3.eth.getBalance(investor);
+
+        investorFinalBalance.sub(investorInitialBalance).should.be.bignumber.equal(amount);
+
+        receipt.logs.length.should.equal(1);
+        receipt.logs[0].event.should.equal('Refunded');
```

Comment:
We have a helper that you should use to test for events:

https://github.com/OpenZeppelin/openzeppelin-solidity/blob/74a62a1314c0c7b68a0d0e7b18b34abdc62b65fa/test/helpers/expectEvent.js#L3-L11

Reply:
(although that might not handle bignumber equality perfectly?) but yeah,

```
await expectEvent.inLogs(receipt.logs, 'Refunded', { investor, weiAmount: amount })
```
should do the trick

Reply:
it indeed does not work for bignumbers, I opened #1026